lib/awsutil/api.go

```diff
@@ -33,6 +33,17 @@ func LoadCredentials() (*CredentialsStore, error) {
 	return loadCredentials()
 }
 
+// TryLoadCredentials works like LoadCredentials but attempts to partially
+// load the credentials in the presence of unloadable accounts. If err is
+// non nil and partial loading succeeded, store will be non-nil and
+// badAccounts will contain the list of accounts that could not be loaded.
+// If err is non nil and partial loading did not succeed then both store
+// and badAccounts will be nil.
+func TryLoadCredentials() (
+	store *CredentialsStore, badAccounts []string, err error) {
+	return tryLoadCredentials()
+}
+
```

Comment:
I don't like this API. I think it's hard to reason about the return conditions. It's also strange to return a non-nil error and yet return something that is (partially) working.
I think it would be cleaner to have an API that returns the credentials store and a slice of error structs, where the error struct has the account name and the error. That's also more useful as the caller can see what errors each account had.